examples/minimal.rs

```diff
@@ -0,0 +1,166 @@
+    // Now we build a cluster parser which takes a script and
+    // an iterator that yields a Token per character
+    let mut parser = Parser::new(
+        Script::Latin,
+        text_content.char_indices().map(|(i, ch)| Token {
+            // The character
+            ch,
+            // Offset of the character in code units
+            offset: i as u32,
+            // Length of the character in code units
+            len: ch.len_utf8() as u8,
+            // Character information
+            info: ch.into(),
+            // Pass through user data
+            data: 0,
+        }),
+    );
+    // Loop over all of the clusters
+    while parser.next(&mut cluster) {
+        // Map all of the characters in the cluster
+        // to nominal glyph identifiers
+        cluster.map(|ch| charmap.map(ch));
+        // Add the cluster to the shaper
+        shaper.add_cluster(&cluster);
+    }
+
+    let mut scale_context = ScaleContext::new();
+
+    let mut run_offset_x: f32 = 0.0;
+
+    shaper.shape_with(|cluster| {
+        cluster.glyphs.iter().for_each(|glyph| {
+            let mut scaler = scale_context
+                .builder(*font)
+                .size(FONT_SIZE)
+                .hint(true)
+                .build();
+            let image = Render::new(&[
+                Source::ColorOutline(0),
+                Source::ColorBitmap(StrikeWith::BestFit),
+                Source::Outline,
+            ])
+            .format(Format::Subpixel)
+            .render(&mut scaler, glyph.id)
+            .unwrap();
+
+            let glyph_image_data = image.data.as_slice();
+
+            let glyph_x_min: i32 = image.placement.left;
+            let glyph_x_max: i32 = image.placement.left + image.placement.width as i32;
+            let glyph_y_min: i32 = image.placement.top;
+            let glyph_y_max: i32 = image.placement.top + image.placement.height as i32;
+
+            for glyph_x in glyph_x_min..glyph_x_max {
+                for glyph_y in glyph_y_min..glyph_y_max {
+                    // Get the value of this pixel
+                    let glyph_byte_offset = (glyph_x - glyph_x_min
+                        + (glyph_y - glyph_y_min) * image.placement.width as i32)
+                        * 4;
+
+                    if glyph_byte_offset < 0 {
+                        eprintln!("Glyph byte offset is less than 0, cannot lookup pixel data");
+                        panic!("Invalid glyph byte offset");
+                    }
+
+                    let glyph_byte_offset = glyph_byte_offset as usize;
+
+                    let pixel_r = 255 - glyph_image_data[glyph_byte_offset];
+                    let pixel_g = 255 - glyph_image_data[glyph_byte_offset + 1];
+                    let pixel_b = 255 - glyph_image_data[glyph_byte_offset + 2];
+
+                    if pixel_r == 255 && pixel_g == 255 && pixel_b == 255 {
+                        // Blank pixel, skip
+                        continue;
+                    }
```

Comment:
Should I switch the renderer to use [`Source::Bitmap`](https://docs.rs/swash/0.1.17/swash/scale/enum.Source.html#variant.Bitmap) instead? Seems like that might let me clean this up.